pkg/shared/util/grpc_utils.go

```diff
@@ -20,18 +20,77 @@ import (
 	"context"
 	"fmt"
 	"log"
+	"strings"
 	"time"
 
+	"github.com/Masterminds/semver/v3"
 	"github.com/numaproj/numaflow-go/pkg/info"
 	"google.golang.org/grpc"
 	"google.golang.org/grpc/codes"
 	"google.golang.org/grpc/credentials/insecure"
 	"google.golang.org/grpc/status"
 
+	"github.com/numaproj/numaflow"
 	sdkerr "github.com/numaproj/numaflow/pkg/sdkclient/error"
 	resolver "github.com/numaproj/numaflow/pkg/sdkclient/grpc_resolver"
 )
 
+func checkConstraint(version *semver.Version, constraint string) error {
+	c, err := semver.NewConstraint(constraint)
+	if err != nil {
+		return fmt.Errorf("error parsing constraint: %w", err)
+	}
+
+	isValid, _ := c.Validate(version)
+	if !isValid {
+		return fmt.Errorf("version did not meet constraint requirement")
+	}
+
+	return nil
+}
+
+// isCompatible checks if the current numaflow version is compatible with the given language's SDK version
+func isCompatible(serverInfo *info.ServerInfo) error {
```

Comment:
instead should I just change the name to checkCompatability instead, since i think returning the bool is redundant when im already using an error to indicate whether the versions are compatible? wdyt?

Reply:
Makes sense